streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1148,6 +1148,17 @@ private static Pattern buildPattern(final Collection<String> sourceTopics,
         return results;
     }
 
+    public List<String> sourceTopicsForStore(final String storeName) {
+        final List<String> results = new ArrayList<>();
+        for (final Map.Entry<String, Set<String>> entry : stateStoreNameToSourceTopics.entrySet()) {
+            if (entry.getKey().equals(storeName)) {
```

Comment:
can't we just lookup the map? why iterate over if we are going to compare the key anyway?

Reply:
I think the return type of stateStoreNameToSourceTopics is a set and we wanted to return a list, so I had to loop and add it to list.

Reply:
I'm also a little confused. Would this not be the same?
```
results.addAll(maybeDecorateInternalSourceTopics(stateStoreNameToSourceTopics.get(storeName))
```
I.e., converting the returned set to a list doesn't seem to require iterating over the whole map just to look for one key.

On a different note, it doesn't seem like we really need a List in particular. We could just relax it to Collection, and then you wouldn't have to convert the set to a list. But that's not a big issue to me.

Reply:
Makes sense, Changed to Collection.